### PR TITLE
Add create snapshots for multiple hosts

### PR DESCRIPTION
**Pull request by m-bucher**
Adds an additional entry in the 'Select Action' select-field on the hosts-page.

### CODE REVIEW

File: app/controllers/foreman_snapshot_management/snapshots_controller.rb [hunk 1]
```diff
@@ -67,8 +72,36 @@ def update
...
+
+    def create_multiple_host
+      data = snapshot_params
+      snapshots_created = 0
+      @hosts.each do |h|
+        s = Snapshot.new(data.merge(host: h))
+        if s.create
+          snapshots_created += 1
+        else
+          error _('Error occurred while creating Snapshot for "%{name}": %{msg}') % { name: h.name, msg: s.errors.full_messages.to_sentence }
```
Comment (mdellweg) [inline app/controllers/foreman_snapshot_management/snapshots_controller.rb]: There was an error with the debian partition table not being available. While this is definitely another issue, it showed, that this error will not be shown. Maybe is were better to use `create!` and handle all failures with `try`.

Comment (timogoebel) [reply app/controllers/foreman_snapshot_management/snapshots_controller.rb]: I would not recommend using try there. Actually, this doesn't look too bad. `Shapshot#create` should return false if there were any errors.
Question is if the in invalid host object should prevent the snapshot from being created in the first place.

File: app/controllers/foreman_snapshot_management/snapshots_controller.rb [hunk 2]
```diff
@@ -67,8 +72,36 @@ def update
...
+        end
+      end
+      if snapshots_created > 0
+        success _('Created %{snapshots} for %{num} %{hosts}') % {
+          snapshots: n_('Snapshot', 'Snapshots', snapshots_created),
+          num: snapshots_created,
+          hosts: n_('host', 'hosts', snapshots_created)
+        }
```
Comment (timogoebel) [inline app/controllers/foreman_snapshot_management/snapshots_controller.rb]: How about adding an explicit return here?

File: app/controllers/foreman_snapshot_management/snapshots_controller.rb (hunk 1 above)
Comment (mdellweg) [reply app/controllers/foreman_snapshot_management/snapshots_controller.rb]: I guess it should not. #26 
Actually it should never be invalid in the first place.
But i guess you suggest that `create` should catch the exception and return `false`?

Comment (timogoebel) [reply app/controllers/foreman_snapshot_management/snapshots_controller.rb]: Either that or the save on the host object could be done with `host.save(validate: false)` to ignore validation errors. But I don't know if that'll actually save the audit record as I believe it also skips callbacks.

File: app/controllers/foreman_snapshot_management/snapshots_controller.rb (hunk 2 above)
Comment (m-bucher) [reply app/controllers/foreman_snapshot_management/snapshots_controller.rb]: I am not sure what you mean.
Is the return not covered by the `redirect_back_or_to`?

**Comment by m-bucher** [comment]
* make sure to show all errors in one flash-message (only one possible at a time)
  I would have liked to make a line-break within the error-flash, but I'm not sure if that is possible.
* added 2 unit-test: valid and invalid bulk-created

File: app/controllers/foreman_snapshot_management/snapshots_controller.rb (hunk 2 above)
Comment (timogoebel) [reply app/controllers/foreman_snapshot_management/snapshots_controller.rb]: I think it's fine in the current version ;-)

**Comment by m-bucher** [comment]
Thanks everyone for your patience :-)

**Pull request merged**